Samples/Tests/Core/IntegrationTestBase.swift

```diff
@@ -0,0 +1,243 @@
+    }
+
+    override func tearDownWithError() throws {
+        try super.tearDownWithError()
+
+        app.terminate()
+        _ = app.wait(for: .notRunning, timeout: appTerminateTimeout)
+
+        if let files = FileManager.default.enumerator(atPath: installUrl.path()) {
+            log.info("Remaining KSCrash files:")
+            for file in files {
+                log.info("\t\(file)")
+            }
+        }
+
+        try? FileManager.default.removeItem(at: installUrl)
+    }
+
+    func launchApp() {
+        app.launch()
+        XCTAssert(
+            sampleButton(.installView(.installButton))
+            .waitForExistence(timeout: appLaunchTimeout)
+        )
+    }
+
+    func sampleButton(_ element: TestElementId) -> XCUIElement {
+        app.buttons[element.accessibilityId]
+    }
+
+    func sampleTextField(_ element: TestElementId) -> XCUIElement {
+        app.textFields[element.accessibilityId]
+    }
+
+    func tapButtons(_ elements: [TestElementId], scrollsLimit: Int = 10) throws {
+        enum Error: Swift.Error {
+            case tooManyScrolls
+        }
+
+        for element in elements {
+            let button = sampleButton(element)
+            if !button.waitForExistence(timeout: screenLoadingTimeout) {
+                var scrollsLeft = scrollsLimit
+                while !button.exists {
+                    if scrollsLeft == 0 {
+                        throw Error.tooManyScrolls
+                    }
+
+                    #if os(watchOS)
+                    app.swipeUp(velocity: .init(floatLiteral: 100.0))
+                    #else
+                    app.swipeUp(velocity: .slow)
+                    #endif
+                    _ = button.waitForExistence(timeout: 0.25)
+                    scrollsLeft -= 1
+                }
+            }
+            button.tap()
+        }
+    }
+
+    func setToggle(_ element: TestElementId, val: Bool) {
+        let uiElement = self.app.switches[element.accessibilityId]
+        XCTAssert(uiElement.waitForExistence(timeout: screenLoadingTimeout))
+        let innerSwitch = uiElement.switches.firstMatch
+        let currentVal = (innerSwitch.value as? String) == "1"
+        if currentVal != val {
+            innerSwitch.tap()
+        }
+    }
+
+    func waitForCrash() {
+        XCTAssert(app.wait(for: .notRunning, timeout: appCrashTimeout), "App crash is expected")
+    }
+
+    private func findRawCrashReportUrl() throws -> URL {
+        enum Error: Swift.Error {
+            case reportNotFound
+        }
+
+        let reportsUrl = installUrl.appending(component: "Reports")
+        let reportUrl = try FileManager.default
+            .contentsOfDirectory(atPath: reportsUrl.path())
+            .first
+            .flatMap { reportsUrl.appending(component:$0) }
+        guard let reportUrl else { throw Error.reportNotFound }
+        return reportUrl
+    }
+
+    func readRawCrashReportData() throws -> Data {
+        enum Error: Swift.Error {
```

Comment:
Just asking. Is it OK to use `Error` which can be ambiguous in some cases? Also, maybe it would be more useful to define one `Error` enum for the whole TestCase?

Reply:
Well, these are only valid inside tests and not expected to be used as API. Made it a bit more clear but not sure if we need a single common enum